### DISCUSSION TITLE
Option to Import & Export in native Mirth's XML format

### DISCUSSION THREAD

**Discussion by gmartelly** [SagaHealthcareIT/mirthsync#51] [Ideas]
Can an option be added where you can import and export channels as fully self contained XML files in the native format that Mirth generates from the GUI? In other words, not separating the channel across multiple javascript files. This would leave the flexibility to manually import or export files from the GUI into the repo instead of having to always run the mirthsync command.

**Comment by carm-afea** [comment]
Definitely a duplicate of #34

**Reply by jessesaga** [reply]
Yeah, I just added a new comment on #34 that has info about a workaround that can be used now and plans for a solution.

**Comment by gmartelly-suki** [comment]
@jessesaga in release 3.1.0  the "items"option for --disk-mode provides the functionality I was looking for. Thank you. I now have the flexibility to export a channel from Mirth directly into the src directory. The only issue is the index.xml file is not updated to reference the manually added channel.

What are your thoughts on having mirthsync have a --refresh flag that checks each directory to see if the number of items in a channel or code template group are less than the number of artifacts in the folder? If that is the case, it can scan the folder and update the index.xml with the newly added ids. This can be done manually by the user or a github action could call it before a commit.

Also, being able to create a new folder in any of the src subdirectories and having the --refresh function automatically generate an index file by generating a group UUID and adding all the artifacts in that directory to it would be great.

**Reply by jessesaga** [reply]
I think that's a great idea. It's related to #26 but is less about Mirth-side changes and more about local target directory changes. We need for Mirthsync to be able to detect and properly deal with renames, additions, and deletions from several different sources. Mirth -> target dir, target-dir -> Mirth and, target-dir only (without interacting with Mirth).

I like the idea of simplifying the large(ish) task of accounting for every add/delete/rename scenario and, for now, just focusing on a 'refresh' scenario where Mirthsync does a sanity check of the target directory without talking to a remote Mirth.

Some aspects of this that warrant more thought....

- Is the verb 'refresh' enough for Mirthsync to know how to properly validate and 'fix' the target directory structure and XML based on local additions, renames, deletions or might we need an additional flag indicating 'add', 'remove', or 'rename'? 

I think we might not need additional information if we make the assumption that file presence and naming trumps 'parent' XML contents but I'm not sure yet.

- Should 'refresh' be another verb alongside 'push' and 'pull'? 

 My initial thought is that it should be.

- Is 'refresh' the best name for this? 

Thinking 'tidy', 'validate', 'fix', 'rebuild'....or maybe something that indicates that a remote Mirth is not relevant like 'refresh-local'. I like 'refresh' and 'rebuild' currently but I'm not very partial to anything at this point...
